fix: update x5c requirements to clarify optionality based on client_id scheme (#1118)
italia/eidas-it-wallet-docs: docs/en/remote-flow.rst

  * - **trust_chain**
    - OPTIONAL. It is a sequence of Entity Statements that composes the Trust Chain related to the Relying Party, as defined in `OID-FED`_ Section 4.3 *Trust Chain Header Parameter*.
  * - **x5c**

>     - REQUIRED when ``client_id`` uses an ``x509_hash`` prefix scheme. OPTIONAL when ``client_id`` uses the ``openid_federation`` scheme. Contains the X.509 certificate chain of the Relying Party, excluding the Trust Anchor certificate, used to verify the JWT signature with the public key in the Relying Party’s certificate as defined in :rfc:`7515`. The Relying Party’s certificate in ``x5c`` asserts the Relying Party identity information along with the network endpoints used in the presentation flow, including the endpoints Authorization Request and Response endpoints (``response_uri`` and ``redirect_uri``). All the endpoints used in the presentation flow by a Relying Party MUST be bound to the FQDN and any further webpath provided in the Relying Party’s certificate, in the form of URI-type SAN for full-URI matching, or a DNSName SAN for host-name matching.

Reviewer: ```suggestion
    - REQUIRED when ``client_id`` uses an ``x509_hash`` prefix scheme. OPTIONAL when ``client_id`` uses the ``openid_federation`` scheme. It contains the X.509 certificate chain about the Relying Party, excluding the Trust Anchor certificate. This certificate MUST be used to verify the JWT signature. The Relying Party’s certificate in ``x5c`` asserts the Relying Party identity information along with the network endpoints used in the presentation flow, including the endpoints Authorization Request and Response endpoints (``response_uri`` and ``redirect_uri``). All the endpoints used in the presentation flow MUST be bound to the FQDN and any further webpath provided in the Relying Party’s certificate, in the form of URI-type SAN for full-URI matching, or a DNSName SAN for host-name matching.
```
